social_navigation_layers/src/human_layer.cpp

```diff
@@ -0,0 +1,223 @@
+//
+//      Human Layer Costmap Plugin Class
+//       by Travis Riggs
+//
+//  This class implements a costmap 2D plugin that represents humans. It
+//  is primarily intended as a layer for the global costmap.
+//
+#include <social_navigation_layers/human_layer.h>
+
+
+namespace social_navigation_layers
+{
+
+// Initializes parameters for the Human Layer
+//
+// The costmap calls this method when it loads this class as a plugin.
+void HumanLayer::onInitialize()
+{
+  SocialLayer::onInitialize();
+  ros::NodeHandle nh("~/" + name_), g_nh;
+  server_ = new dynamic_reconfigure::Server<HumanLayerConfig>(nh);
+  f_ = boost::bind(&HumanLayer::configure, this, _1, _2);
+  server_->setCallback(f_);
+
+  lethal_cost_ = (double) costmap_2d::LETHAL_OBSTACLE;
```

Comment:
nit: there is no need to store lethal_cost_ or inscribed_cost_ in the object as they are constants. You could make these just `const double lethal_cost = static_cast<double>(costmap_2d::LETHAL_OBSTACLE);` in the places you need the constants. The compiler can then optimize it away entirely. Alternatively, you could make them static const members but the syntax for those is sometimes tricky. If we are using modern C++, you can probably just specify them straight in the include file. Also, I would not use C style casts anywhere, use `static_cast<double>()`

Reply:
I was mainly concerned about the readability of the code.  Casting them every time they were used made the logic hideous.

I tried a few of your suggestions.  I settled on using `inline static const` in the header.  But the compiler also allowed me to just declare them there as `static`.

I also specified C++17 in CMakeLists.

What do you think?

Reply:
Also, just for my learning, I guess C-style casts are bad in C++?  Does static_cast perform more checking?  Or does it do some optimization?